zim/tmp-kiwix/cleanup-old-files.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/local/bin/python3
+
+"""
+File and Directory Deletion Script
+
+This script performs the deletion of files older than a specified number of days and
+deletes empty subdirectories in a specified folder.
+
+Usage:
+    python cleanup-old-files.py -f /path/to/folder -n 30 [-d]
+
+Options:
+    -d, --dry-run   Perform a dry run to list the files and directories that would be deleted.
+                    If not specified, the script performs the actual deletion.
+    -f, --folder    Path to the target folder.
+    -n, --days      Delete files older than this many days. Default is 30 days.
+"""
+
+import os
+import argparse
+import datetime
+
+def list_files_to_delete(folder_path, days):
+    """
+    List files that are older than a specified number of days.
+
+    Args:
+        folder_path (str): The path to the target folder.
+        days (int): Delete files older than this many days.
+
+    Returns:
+        list: A list of file paths to be deleted.
+    """
+    current_time = datetime.datetime.now()
+    files_to_delete = []
+
+    def list_files(folder_path):
+        for foldername, subfolders, filenames in os.walk(folder_path):
+            for filename in filenames:
+                file_path = os.path.join(foldername, filename)
+                yield file_path
+
+    for file_path in list_files(folder_path):
+        # Get the last modification time of the file
+        file_time = datetime.datetime.fromtimestamp(os.path.getmtime(file_path))
+
+        # Calculate the difference in days
+        days_difference = (current_time - file_time).days
+
+        if days_difference > days:
+            files_to_delete.append(file_path)
+
+    return files_to_delete
+
+def list_directories_to_delete(folder_path, files_to_delete):
+    """
+    List empty subdirectories or subdirectories containing only files to be deleted.
+
+    Args:
+        folder_path (str): The path to the target folder.
+        files_to_delete (list): A list of file paths to be deleted.
+
+    Returns:
+        list: A list of empty subdirectories to be deleted.
+    """
+    empty_directories_to_delete = []
```

Comment:
I'd probably loop on files_to_delete's parent recursively to build a list of paths to delete